ci/release.py

```diff
@@ -62,79 +73,121 @@ def check_git_clean():
         raise RuntimeError(f"git status is not clean:\n{git_status.stdout}")
 
 
-def get_current_go_version_from_changelog(changelog_content: str):
-    match = re.search(r"modal-go/v(?P<major>[\d]+)\.(?P<minor>[\d]+)\.(?P<patch>[\d]+)", changelog_content)
-    if not match:
-        raise RuntimeError("Unable to parse modal-go version")
-    current_go_verison = {key: int(match.group(key)) for key in ["major", "minor", "patch"]}
-    return current_go_verison
+def get_current_js_version():
+    package_path = Path("modal-js") / "package.json"
+    with package_path.open("r") as f:
+        json_package = json.load(f)
+        return json_package["version"]
 
 
 def update_version(args):
-    """Updates version and changelog to prepare for release.."""
+    """Updates version and changelog and prepare a release PR."""
     if args.update not in ["major", "minor", "patch"]:
         raise RuntimeError("update parameter must be 'major', 'minor', or 'patch'")
 
-    # Make sure changelog has new items in "Unreleased"
-    changelog_path = Path("CHANGELOG.md")
-    changelog_content = changelog_path.read_text()
-    check_unreleased_has_items(changelog_content)
-
     check_git_clean()
 
-    # Get updated go version
-    go_version = get_current_go_version_from_changelog(changelog_content)
-    go_version[args.update] += 1
-    new_go_version = f"v{go_version['major']}.{go_version['minor']}.{go_version['patch']}"
+    if args.dev:
+        current_version = get_current_js_version()
 
-    # Update and get new js version
-    run_cli(["npm", "version", args.update], text=True, cwd="modal-js")
-    package_path = Path("modal-js") / "package.json"
-    with package_path.open("r") as f:
-        json_package = json.load(f)
-        new_js_version = json_package["version"]
+        if "-dev." in current_version:
+            run_cli(["npm", "version", "prerelease", "--no-git-tag-version"], cwd="modal-js")
+        else:
+            run_cli(["npm", "version", f"pre{args.update}", "--preid=dev", "--no-git-tag-version"], cwd="modal-js")
+
+        new_version = get_current_js_version()
+
+        run_cli(["git", "diff"])
 
-    # Update changelog with versions
-    version_header = f"modal-js/v{new_js_version}, modal-go/{new_go_version}"
+        commit_message = f"[DEV-RELEASE] Prepare dev release for modal-js/v{new_version}, modal-go/v{new_version}"
+        if args.dry_run:
+            print("\nDRY RUN: Would create commit with message:")
+            print(commit_message)
+            run_cli(["git", "restore", "--", "modal-js/package.json", "modal-js/package-lock.json"])
```

Comment:
Nit: If `npm version ...` a few lines above were to fail, then this `git restore` would not run, leaving the files modified.